webbpsf/opds.py

```diff
@@ -3207,8 +3206,76 @@ def coeffs_to_seg_state(coeffs):
     seg_state[:,1] = coeffs_tab[:,2]  # tilt in 2nd column
     return seg_state*1e6   # convert from meters & radians to micro units
 
+
+@functools.lru_cache
+def _get_lom(npix):
+    """Initialize a few things that are computationally expensive so we cache for multiple reuses"""
+
+    ote_lom = OTE_Linear_Model_WSS(npix=npix)
+    jw_ptt_basis = JWST_WAS_PTT_Basis()
+    return ote_lom, jw_ptt_basis
+
+
+def decompose_opd_segment_PTT(opd, plot=False):
```

Comment:
I tried using this function (`decompose_opd_segment_PTT`) by expanding the example in https://webbpsf.readthedocs.io/en/latest/jwst_measured_opds.html
For this, after loading the OPD, I inspected and used it with the decomposed function. 
This is how the OPD looks, `plt.imshow(nrc.pupilopd[0].data)`: 
![image](https://user-images.githubusercontent.com/9635910/186296169-ece39524-99f6-4d60-8072-0b7121d212f9.png)
Then I applied the function as:
`webbpsf.opds.decompose_opd_segment_PTT(nrc.pupilopd[0].data, plot = True)`
and I get:
(array([[nan, nan, nan, ..., nan, nan, nan],
        [nan, nan, nan, ..., nan, nan, nan],
        [nan, nan, nan, ..., nan, nan, nan],
        ...,
        [nan, nan, nan, ..., nan, nan, nan],
        [nan, nan, nan, ..., nan, nan, nan],
        [nan, nan, nan, ..., nan, nan, nan]]),
 array([ 7.99506642e-09,  1.55340648e-08, -1.45810110e-08, -1.64441644e-08,
        -1.77063423e-09, -1.91410874e-08, -2.05439222e-08, -2.98547854e-09,
        -8.20687819e-09, -4.84058661e-09,  1.16624158e-08,  3.19651998e-08,
         1.52231422e-08,  1.86759131e-08, -1.61536156e-08,  2.21041071e-09,
         5.61439423e-08,  1.81762701e-08, -3.43559173e-09, -1.04949073e-08,
        -1.57846742e-08,  1.88472787e-09,  8.24705443e-09, -9.52822306e-09,
         1.90445152e-08,  2.29304796e-08, -2.46303600e-08, -1.12180320e-08,
        -7.22103361e-09, -1.77367429e-08,  2.62729490e-09,  3.92728519e-09,
         1.12543984e-08, -7.95697084e-09,  4.30708180e-09, -8.53238522e-09,
         5.46708572e-09, -5.92302208e-09,  1.57581015e-09,  5.59286666e-09,
        -5.79568266e-09, -3.81148890e-10,  8.19012707e-09,  1.14422039e-09,
         1.23863254e-08, -2.30900109e-09, -4.10657550e-09, -9.72345849e-09,
         2.31787969e-09,  6.65293229e-09,  9.50394333e-09, -1.10150228e-08,
        -5.89912643e-09, -1.47712434e-08]))

![image](https://user-images.githubusercontent.com/9635910/186296294-d06d474a-3eb7-4bf7-8eea-96ad00751a53.png)

The values seems reasonable, at least in terms of range of expected corrections, but the figure seems out of scale. Maybe it needs a better/dynamic scaling range for display.

Reply:
I think the display is getting confused about units, yeah. Looks like the default scaling range is expecting microns but the array values are in meters

Reply:
I made the default better, and also allowed the user to change the vmax if desired: 

<img width="919" alt="Screen Shot 2022-09-14 at 5 38 44 PM" src="https://user-images.githubusercontent.com/1151745/190267489-4efb8aab-2d93-4871-9f79-c5762ba2d4d2.png">